src/de/uni_passau/fim/gitwrapper/StateReason.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (C) 2025 Shiraz Jafri
+ *
+ * This file is part of GitHubWrapper.
+ *
+ * GitHubWrapper is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * GitHubWrapper is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with GitWrapper. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.uni_passau.fim.gitwrapper;
+
+/**
+ * Enumeration of reasons for the state of an Issue or PullRequest.
+ */
+public enum StateReason {
+    /**
+     * Denotes that the Issue or PullRequest was completed.
+     */
+    COMPLETED,
+
+    /**
+     * Denotes that the Issue or PullRequest was reopened.
+     */
+    REOPENED,
+
+    /**
+     * Denotes that the Issue or PullRequest was not planned.
+     */
+    NOT_PLANNED,
+
+    /**
+     * Denotes that the Issue or PullRequest was a duplicate.
+     */
+    DUPLICATE,
+
+    /**
+     * Denotes that the state reason is not known or not specified.
+     * This can be used when the state reason is not applicable or not provided.
+     */
+    NONE,
+
+    /**
+     * Any state reason is included (can be used for unknown reasons).
+     */
+    ANY;
```

Comment:
I remember that we had a discussion about what the difference between `NONE` and `ANY` is. I don't remember the outcome. Could you please check. Maybe we need to remove one of them...

And how do these enums transfer to the outcome in the resulting json files? Let's discuss that in our next meeting.